apps/shared/avifpng.c

```diff
@@ -396,11 +407,16 @@ avifBool avifPNGWrite(const char * outputFilename, const avifImage * avif, uint3
 
     volatile int rgbDepth = requestedDepth;
     if (rgbDepth == 0) {
-        if (avif->depth > 8) {
-            rgbDepth = 16;
+#if defined(AVIF_ENABLE_EXPERIMENTAL_YCGCO_R)
+        if (avif->matrixCoefficients == AVIF_MATRIX_COEFFICIENTS_YCGCO_RE ||
+                                    avif->matrixCoefficients == AVIF_MATRIX_COEFFICIENTS_YCGCO_RO) {
+            rgbDepth = avif->depth - ((avif->matrixCoefficients == AVIF_MATRIX_COEFFICIENTS_YCGCO_RE) ? 2 : 1);
```

Comment:
IMPORTANT: For PNG, `rgbDepth` can only be 8 or 16. Since `avif->depth` is 8, 10, or 12, the only legal combination here is `avif->depth == 10`, `avif->matrixCoefficients == AVIF_MATRIX_COEFFICIENTS_YCGCO_RE`, and `rgbDepth == 8`.

I will let you work out the details.

Reply:
Done